src/mctpd.c

```diff
@@ -2734,19 +2818,150 @@ static int bus_endpoint_get_prop(sd_bus *bus,
 	return rc;
 }
 
+static int bus_link_get_prop(sd_bus *bus,
+		const char *path, const char *interface, const char *property,
+		sd_bus_message *reply, void *userdata, sd_bus_error *berr)
+{
+	ctx *ctx = userdata;
+	char *tmpstr = NULL;
+	char *link_name = NULL;
+	linkmap_entry *lm = NULL;
+	int rc = 0;
+
+	if (!is_interfaces_path(path)) {
+		sd_bus_error_setf(berr, SD_BUS_ERROR_INVALID_ARGS,
+				"Invalid Object Path");
+		goto out;
+	}
+
+	rc = sd_bus_path_decode_many(path, MCTP_DBUS_PATH "/%/%", &tmpstr,
+		&link_name);
+	if (rc <= 0) {
+		sd_bus_error_setf(berr, SD_BUS_ERROR_INVALID_ARGS,
+				"Invalid Object Path");
+		goto out;
+	}
+
+	lm = find_link_by_name(ctx, link_name);
+	if (!lm) {
+		rc = -ENOENT;
+		goto out;
+	}
+	dfree(tmpstr);
+	dfree(link_name);
```

Comment:
do these need to be deferred?

Reply:
Can you give me some guideline when dfree() is need in mctpd.c?
I just copy/parse from other function.

Reply:
`dfree` is a utility for a "deferred free" of the pointer; the pointer is valid until we return to the dbus event handler, and will be freed during that processing.

In this case, it looks like we don't need those pointers for anything else, so we can free them immediately with regular `free`.

Reply:
I updated the code.